app/models/collection.client.ts

```diff
@@ -1,5 +1,78 @@
 import { runQuery } from "./utils";
 
+export const getAllCollections = async () => {
+  const getAllCollectionsQuery = `{
+    collections(first: 20) {
+      edges {
+        node {
+          title
+          products(first: 1) {
+            edges {
+              node {
+                id
+                images(first: 1) {
+                  edges {
+                    node {
+                      altText
+                      src
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  `;
+
+  const params = {
+    query: getAllCollectionsQuery,
+    variables: {},
```

Comment:
lets create a default value inside the `runQuery` function for variable `{}` so you don't have to pass it in case we don't need it

Reply:
Add also Optional for "Variables"